service/iot-service-samples/azure-sas-credential-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/AzureSasCredentialSample.java

```diff
@@ -96,55 +86,6 @@ public static void main(String[] args)
         runDeviceMethodClientSample(iotHubHostName, credential, newDeviceId);
     }
 
-    /**
-     * Create a shared access signature for a given resource and key. This method isn't used by the sample,
-     * but it is a good reference for how to create shared access signatures in Java.
-     *
-     * @param resourceUri The resource that the shared access token should grant access to. For cases where the token
-     * will be used for more than one function (i.e. used by registryManager to create a device and used by serviceClient
-     * to send cloud to device messages), this value should be the hostName of your IoT Hub
-     * ("my-azure-iot-hub.azure-devices.net" for example). Shared access signatures do support scoping of the resource
-     * authorization by making this resourceUri more specific. For example, a resourceUri of "my-azure-iot-hub.azure-devices.net/devices"
-     * will make this token only usable when creating/updating/deleting device identities.
-     * For more examples, see <a href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#use-security-tokens-from-service-components">Using security tokens from service components</a>
-     * @param key The shared access key for your IoT Hub.
-     * @return The shared access signature. This value can be used as is to build a {@link AzureSasCredential} instance
-     * @throws UnsupportedEncodingException If UTF_8 is not a supported encoding on your device.
-     * @throws NoSuchAlgorithmException If HmacSHA256 algorithm isn't found.
-     * @throws InvalidKeyException If initializing the HmacSHA256 SHA fails.
-     * @see <a href="docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security">Control access to IoT Hub</a>
-     */
-    @SuppressWarnings("unused")
-    public static String generateSharedAccessSignature(String resourceUri, String key)
```

Comment:
This helper function has been moved to the readme since it wasn't actually used in this sample

Reply:
I am not against adding it to readme but wondering if it is better to leave it here for refactoring, updates, etc and add code snippets instead?

Reply:
I like moving this function into the readme because the function isn't used in the sample. Even with the suppressed warning, my fear is that someone will delete it because they see it isn't used, and don't understand that it is there only for reference purposes. This move also keeps the sample itself a bit shorter to read, which is better for users who already know how to build a shared access signature

Reply:
All you have to do is add a comment that says why it is there, and note that it isn't referenced.